adlfs/spec.py

```diff
@@ -584,10 +584,16 @@ def do_connect(self):
                 self.service_client = AIOBlobServiceClient.from_connection_string(
                     conn_str=self.connection_string
                 )
-            elif self.account_name:
-                self.account_url: str = (
-                    f"https://{self.account_name}.blob.core.windows.net"
-                )
+            elif self.account_name is not None:
+                if hasattr(self, 'account_host'):
```

Comment:
Can you help me understand where this `account_host` attribute coming from? Is this from a subclass?

I'd rather have the `__init__` method accept something like `account_host` and make it a property on `AzureBlobFileSystem`. Then we can avoid all the hasattrs.

The interaction with the current `account_name` will be a bit complicated. Can you share an example of how account_host is being specified? Maybe we just don't allow users to pass both.

Reply:
Adlfs is always appending the public blob endpoint. so having the ability to pass the account_host will help in accessing the non-public regions like gov regions. As account_host is not mandatory I was thinking of consumers to pass as required. We can also add it in __init__ and initialize it as None making as not a required property.